scanpipe/models.py

```diff
@@ -2889,6 +2889,18 @@ def walk(self, topdown=True):
             if not topdown:
                 yield child
 
+    def extracted_to(self, codebase=None):
+        """Return the path this Resource archive was extracted to or None."""
+        extract_path = f"{self.path}-extract"
+        return self.project.get_resource(extract_path)
+
+    def extracted_from(self, codebase=None):
```

Comment:
Are the `codebase` args required?

Reply:
Yes, this is to have the same function arguments as the functions in commoncode, so any code written in scancode-toolkit (like in package assembly, which is run in both sctk/scio context) works here too. See the functions at commoncode: https://github.com/aboutcode-org/commoncode/blob/main/src/commoncode/resource.py#L1265

See https://github.com/aboutcode-org/scancode.io/blob/add-resource-methods/scanpipe/models.py#L2826 for other functions in the CodebaseResource model which have the same codebase arg for the same reason.